app/react/kubernetes/applications/CreateView/application-services/ingress/AppIngressPathsForm.tsx

```diff
@@ -0,0 +1,171 @@
+
+import { FormError } from '@@/form-components/FormError';
+import { Button } from '@@/buttons';
+import { SwitchField } from '@@/form-components/SwitchField';
+import { TextTip } from '@@/Tip/TextTip';
+import { Link } from '@@/Link';
+
+import { ServicePortIngressPath } from '../types';
+
+import { AppIngressPathForm } from './AppIngressPathForm';
+
+type Props = {
+  servicePortIngressPaths?: ServicePortIngressPath[];
+  onChangeIngressPaths: (ingressPath: ServicePortIngressPath[]) => void;
+  namespace?: string;
+  ingressPathsErrors?: FormikErrors<ServicePortIngressPath[]>;
+  serviceIndex: number;
+  portIndex: number;
+  isEditMode?: boolean;
+};
+
+export function AppIngressPathsForm({
+  servicePortIngressPaths,
+  onChangeIngressPaths,
+  namespace,
+  ingressPathsErrors,
+  serviceIndex,
+  portIndex,
+  isEditMode,
+}: Props) {
+  const environmentId = useEnvironmentId();
+  const ingressesQuery = useIngresses(
+    environmentId,
+    namespace ? [namespace] : undefined
+  );
+  const { data: ingresses } = ingressesQuery;
+  const ingressControllersQuery = useIngressControllers(
+    environmentId,
+    namespace
+  );
+  const { data: ingressControllers } = ingressControllersQuery;
+
+  // if some ingress controllers are restricted by namespace, then filter the ingresses that use allowed ingress controllers
+  const allowedIngressHostNameOptions = useMemo(() => {
+    const allowedIngressClasses =
+      ingressControllers
+        ?.filter((ic) => ic.Availability)
+        .map((ic) => ic.ClassName) || [];
+    const allowedIngresses =
+      ingresses?.filter((ing) =>
+        allowedIngressClasses.includes(ing.ClassName)
+      ) || [];
+    return allowedIngresses.flatMap((ing) =>
+      ing.Hosts?.length
+        ? ing.Hosts.map((host) => ({
+            label: `${host} (${ing.Name})`,
+            value: host,
+            ingressName: ing.Name,
+          }))
+        : []
+    );
+  }, [ingressControllers, ingresses]);
+
+  if (ingressesQuery.isError || ingressControllersQuery.isError) {
+    return <FormError>Unable to load ingresses.</FormError>;
+  }
+
+  if (ingressesQuery.isLoading || ingressControllersQuery.isLoading) {
+    return <p>Loading ingresses...</p>;
+  }
+
+  return (
+    <div className="flex flex-col gap-y-4">
+      <div className="!mb-0 flex w-full flex-wrap items-center gap-x-4 gap-y-2">
+        <SwitchField
+          fieldClass="w-max gap-x-8"
+          label="Expose via ingress"
+          tooltip="Expose this ClusterIP service externally using an ingress. This will create a new ingress path for the selected ingress hostname."
+          labelClass="w-max"
+          name="publish-ingress"
+          checked={!!servicePortIngressPaths?.length}
+          onChange={(value) => {
+            const newIngressPathsValue = value
+              ? [
+                  {
+                    Host: allowedIngressHostNameOptions[0]?.value ?? '',
```

Comment:
allowedIngressHostNameOptions will never be undefined?

Reply:
Yeah, in the worst case it'll be an empty array